### ISSUE TITLE
Smarty capture BC break since 3.1.28

### ISSUE THREAD

**Issue by GodLesZ** [smarty-php/smarty#153]
Since `3.1.28` Smarty `capture` works different and breaks existing code, again. 
Also happend with `Smarty::getVariable()`, which was marked as deprecated and damaged at the same time (fixed since `3.1.29` but anyways).

# Example for `capture` problem

*layout.tpl*
```
{block "content"}{/block}
```

*index.tpl*
```
{block "content"}
    {include "test-plugin.tpl"}
    {$smarty.capture.plugin}
{/block}
```

*test-plugin.tpl*
```
{capture "plugin"}
    test
{/capture }
```

*index.php*
```
$smarty->display('extends:layout.tpl|index.tpl');
```

The captured content is not available in the index.tpl.
If I move the `capture` tags to the index.tpl, everything works fine. So it seems to have a local scope to the current template now but no `scope` parameter like the `assign` tag.

Any chance to get
1. BC in the future? 3.1.*28* is a patch/build number
2. a `scope` parameter or some advice how to accomplish the `capture` usage across multiple templates?

**Comment by pvenakis** [comment]
+1, `capture` defined in included files is no longer visible in the main scope

**Comment by catherinek** [comment]
+1, `{capture}` no longer works as assigned values in 3.1.29.

````smarty
{capture assign="tpl_footer_text" name="tpl_footer_text"}
	Footer Text
{/capture}
````

`{$tpl_footer_text}` is empty.

**Comment by redemption** [comment]
This still affects 3.1.29 for us.  The capture is not useable in the main scope.

**Comment by GodLesZ** [comment]
@redemption, I meant an Error which was introduced in `3.1.27` on `$smarty->getVariable()` usage, not the `{capture}` behavior. ;)
`$smarty->getVariable()` was marked as deprecated in `3.1.28`, changed slightly and throwed an error until `3.1.29`

**Comment by redemption** [comment]
I just want to add that before this bug, it was possible to use capture content from a separate fetch.

Eg in the @GodLesZ original post:

index.tpl

```
{block "content"}
   {$content}
    {$smarty.capture.plugin}
{/block}
```

test-plugin.tpl

```
Content
{capture "plugin"}
    test
{/capture }
```

index.php

```
$content = $smarty->fetch('test-plugin.tpl');
$smarty->assign('content',$content);
$smarty->display('index.tpl');
```


Output of above would be the following:

Content
test

**Comment by redemption** [comment]
I was wondering, is this something that can be fixed and is intended to be fixed?  Should we rewrite all uses of capture and stop using it?

**Comment by uwetews** [comment]
The fix is now in the master branch and will late be included in 3.1.30
The capture content is now also cached in the Smarty object and so available across several Smarty::fetch() call.

Sorry that it took so long, I was longer time in hospital.

**Comment by pvenakis** [comment]
Hey @uwetews we are very sorry to hear that you had health issues, we wish you a quick recovery! Take care of yourself!

**Comment by pnikolov** [comment]
I get the same problem in the latest _master-dev_ version. The _capture_ is not visible outside the template it is defined.

**Comment by uwetews** [comment]
Sorry, original error was introduced by mistake again.
The fix is now in the master branch and will later be included in 3.1.30
